pkg/proc/bininfo.go

```diff
@@ -442,6 +454,53 @@ func (bi *BinaryInfo) findCompileUnit(pc uint64) *compileUnit {
 
 // ELF ///////////////////////////////////////////////////////////////
 
+func (bi *BinaryInfo) LoadSeparateDebugInfo(exe *elf.File) (*os.File, *elf.File, error) {
+	buildid := exe.Section(".note.gnu.build-id")
+	if buildid == nil {
+		return nil, nil, errors.New("can't find build-id note")
+	}
+
+	br := buildid.Open()
+	bh := new(BuildIdHeader)
+	if err := binary.Read(br, binary.LittleEndian, bh); err != nil {
+		return nil, nil, errors.New("can't read build-id header: " + err.Error())
+	}
+
+	name := make([]byte, bh.Namesz)
+	if err := binary.Read(br, binary.LittleEndian, name); err != nil {
+		return nil, nil, errors.New("can't read build-id name: " + err.Error())
+	}
+
+	if strings.TrimSpace(string(name)) != "GNU\x00" {
+		return nil, nil, errors.New("invalid build-id signature")
+	}
+
+	descBinary := make([]byte, bh.Descsz)
+	if err := binary.Read(br, binary.LittleEndian, descBinary); err != nil {
+		return nil, nil, errors.New("can't read build-id desc: " + err.Error())
+	}
+	desc := hex.EncodeToString(descBinary)
+
+	debugPath := fmt.Sprintf("/usr/lib/debug/.build-id/%s/%s.debug", desc[:2], desc[2:])
+	sepFile, err := os.OpenFile(debugPath, 0, os.ModePerm)
+	if err != nil {
+		return nil, nil, errors.New("can't open separate debug file: " + err.Error())
```

Comment:
This error and all the ones that follow should include the path of the debug file (printed as `%q`), so that we can tell for sure which file we are trying to read.

Reply:
The error coming from os.OpenFile already includes the path. I've added it to the other two.

Reply:
Thanks.